app/src/main/java/net/gsantner/markor/util/DocumentIO.java

```diff
@@ -132,27 +162,47 @@ public static synchronized boolean saveDocument(final Document document, final S
         document.setFile(documentInitial.getFile());
 
         if (!text.equals(documentInitial.getContent())) {
-            document.forceAddNextChangeToHistory();
-            document.setContent(text + (!TextUtils.isEmpty(text) && !text.endsWith("\n") ? "\n" : ""));
+            ret = writeContent(document, text, shareUtil, context);
+        } else {
+            ret = true;
+        }
+        return ret;
+    }
+
+    private static boolean writeContent(Document document, String text, ShareUtil shareUtil, Context context) {
+        boolean ret;
+        document.forceAddNextChangeToHistory();
```

Comment:
what are these two lines required for?

Reply:
174&175 was added by you 
https://github.com/gsantner/markor/commit/247155b1054eb50d903d9046f224b1d98df68854
https://github.com/gsantner/markor/commit/90c929a6efa74b4f5c63f20ad9a3cad931fd42ad
I only extract methods and added the boolean ret.
See https://github.com/gsantner/markor/commit/b1b2dca421ef0e544b963603202309b802516dd1
this was enforced by the automatic review tool

Reply:
Oh I see, well well